staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -165,6 +157,25 @@ func (h *DeploymentHistoryViewer) ViewHistory(namespace, name string, revision i
 	})
 }
 
+// GetHistory returns the ReplicaSet revisions associated with a Deployment
+func (h *DeploymentHistoryViewer) GetHistory(namespace, name string) (map[int64]runtime.Object, error) {
+	allRSs, err := getDeploymentReplicaSets(h.c.AppsV1(), namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[int64]runtime.Object)
+	for _, rs := range allRSs {
+		v, err := deploymentutil.Revision(rs)
+		if err != nil {
+			continue
```

Comment:
Why we continue in here?. As far as I understand, error in here is not a fatal thing?

Reply:
I think I agree that it should return an error.  I was following the way that ViewHistory did it (see line 114 above), but it does seem that it would be misleading if it did not actually fail if there was an error.

Let me see about changing this to return an error...

Reply:
I think I'm going to leave this as continue for now, and have added warning messages if it skips a replicaset.

It looks like this was done intentionally in case it is unable to parse the revision from the replicaset's annotation, so that you never get into a situation where you can't see the rollout history because of a single corrupt replicaset's metadata.

I could be convinced that returning an error is the correct thing to do, it just gives me pause to change this since this is how it was already being done.  I wish there was a comment here or something explaining why continue is being used.  At least there will be a warning now.